### PR TITLE
Output gradientstops in serialization and debug output

### LINKED ISSUE #2328: SetGradientStops missing with debug-serialization
Hi,
I found that serializing a BuiltDisplayList with the "debug-serialization" feature omits the `SetGradientStops` items.

BTW I am curious to hear why the gradient stops are separate items instead of being part of gradient and radial gradient.

* [serialized display list](https://gist.github.com/pyfisch/29af74e0ebffc2d6287ef806143c3cd4)
* Rendering: simple-gradient

cc @kvark

### PR DESCRIPTION
**Pull request by Gankro**
Fixes #2328 (I think?)

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/2709)

### CODE REVIEW

File: webrender_api/src/display_list.rs [hunk 1]
```diff
@@ -236,35 +236,46 @@ impl<'a> BuiltDisplayListIter<'a> {
         self.cur_clip_chain_items = ItemRange::default();
 
         loop {
-            if self.data.is_empty() {
-                return None;
+            self.next_raw();
+            if let SetGradientStops = self.cur_item.item {
+                // SetGradientStops is a dummy item that most consumers should ignore
```
Comment (kvark) [inline webrender_api/src/display_list.rs]: can we leave this decision to the clients, given that they do know which items they need and which they don't?

**Comment by kvark** [comment]
I do confirm the PR fixes the issue with capturing gradients :+1: :tada:

File: webrender_api/src/display_list.rs (hunk 1 above)
Comment (Gankro) [reply webrender_api/src/display_list.rs]: They can just use `_raw` if they want to make that decision, right? This method is for the "intended" case, where SetGradientStops is really just a quirk of our encoding.

**Pull request merged**